c/src/demo_masq_ds.c

```diff
@@ -0,0 +1,199 @@
+
+
+/* 
+* Initialize/set seed for any random number generator used
+* We recommand to call QiSpace API (i.e. QE) to get seed 
+* if no good quantum safe level entropy source
+*/
+int32_t seed_init()
+{
+    /* here we just use std lib rand to set seed for demo purpose */
+    srand(time(0));
+    for(int j = 0; j < seed_len; j++){
+            seed_orig[j] = rand() & 0xff;
+    }
+    return 1;
+}
+
+/* Customize the random callback function below according to specific needs. 
+ * Otherwise, default random CF is Quantropi 
+ */
+int32_t rand_cf(MASQ_RAND_handle rand_handle, int32_t rand_length,  uint8_t *rand_num){
+    
+    if (rand_num == NULL) return 0;
+    
+    if (rand_handle == NULL) 
+    {
+        for (int i = 0; i < rand_length; i++)
+        {
+            rand_num[i] = rand() & 0xff;
+        }
+    }
+    return 1;
+}
+
+int32_t rand_seed_cf(MASQ_RAND_handle rand_handle, uint8_t *seed, int32_t seed_length)
+{
+    if (seed == NULL) return 0;
+
+    if(rand_handle == NULL){
+        seed = seed_orig;
+    }
+    return 1;
+}
+
+static unsigned char *rand_string(unsigned char *str, size_t size)
+{
+    const unsigned char charset[] = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJK0123456789";
+    if (size) {
+        --size;
+        for (size_t n = 0; n < size; n++) {
+            int key = rand() % (int) (sizeof charset - 1);
+            str[n] = charset[key];
+        }
+        str[size] = '\0';
+    }
+    return str;
+}
+
+
+void sendPubKey(){
+    printf("\n*Alice sends Bob her public key and signature..*\n");
+}
+
+int main(int argc, const char * argv[]) {
+
+    MASQ_DS_handle *ds_handle_alice, *ds_handle_bob;
+    int32_t len_pk_alice, len_sk_alice, len_signature;
+    unsigned char *pk_alice, *sk_alice, *signature;
+    int len_msg = 1024;
+    unsigned char *msg = malloc(len_msg + 1);
+    if(msg)
+    {
+        /* Generating random message */
+        rand_string(msg, len_msg);
+    }
+
+    /* Initialize random seed */ 
+    seed_init();
+
+    /* Init: Initialize MASQ DS handles for Alice and Bob */ 
+    printf("\nInitialize MASQ DS handle for Alice and Bob...\n");
+
+    /* Uses default Quantropi random callback function */
+    ds_handle_bob = MASQ_DS_init(MPPKL5, NULL, NULL, NULL); 
+
+    /* Uses provided random callback function */
+    //ds_handle_bob = MASQ_DS_init(MPPKL5, rand_cf, rand_seed_cf, NULL); 
+    
+    if(ds_handle_bob == NULL) {
+        printf("\nMASQ DS Handle init failed for Bob.\n");    
+        return MASQ_ERR_MEM_Alloc;    
+    }
+    
+    /* Uses default Quantropi random callback function */
+    ds_handle_alice = MASQ_DS_init(MPPKL5, NULL, NULL, NULL); 
+    
+    /* Uses provided random callback function */
+    //ds_handle_alice = MASQ_DS_init(MPPKL5, rand_cf, rand_seed_cf, NULL);  
+    
+    if(ds_handle_alice == NULL) {
+        printf("\nMASQ DS Handle init failed for Alice.\n");    
+        return MASQ_ERR_MEM_Alloc;    
+    }
+    printf("\nSuccess!\n");
+
+
```

Comment:
generate rand seed for Bob:  gen_rand_seed_bob()  -> buffer: bob_seed[1024]
generate rand seed for Alice: gen_rand_seed_alice() -> buffer:  alice_seed[1024]